src/Nethermind/Nethermind.Blockchain/BlockTree.cs

```diff
@@ -887,6 +850,81 @@ private void DeleteBlocks(Hash256 deletePointer)
             }
         }
 
+        public IEnumerable<Block> DeleteBlocksBeforeTimestamp(ulong cutoffTimestamp, CancellationToken cancellationToken)
+        {
+            BlockAcceptingNewBlocks();
+            List<Block> blocks = [];
```

Comment:
Storing the blocks in a list is probably not a great idea as this might accidentally load a lot of blocks on existing node.